src/posix/platform/radio.cpp

```diff
@@ -978,6 +978,15 @@ otError otPlatRadioReceiveAt(otInstance *aInstance, uint8_t aChannel, uint32_t a
     return OT_ERROR_NOT_IMPLEMENTED;
 }
 
+#if OPENTHREAD_CONFIG_PLATFORM_BOOTLOADER_MODE_ENABLE
+void otPlatResetToBootloader(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    IgnoreError(GetRadioSpinel().SendReset(SPINEL_RESET_BOOTLOADER));
```

Comment:
We have some guideline about compatibility between host and RCP.
https://github.com/openthread/openthread/blob/6ff277841ebd6045be87d8407c840df9bc35709c/src/lib/spinel/spinel.h#L311-L372

We should consider impact of this change:

- If host has `BOOTLOADER_MODE_ENABLE`, it sends "reset" command with this new reset status `SPINEL_RESET_BOOTLOADER`
- An older RCP (or one that does not support the `BOOTLOADER_MODE_ENABLE`), will ignore this new reset status and performs normal reset.
- This can be confusing (people may ask why this does not work when I enable it, forgetting that it needs to be enabled not just on host but on RCP as well.

----

Adding @zhanglongxia  here as well. 

Our goal is to implement any new RCP/host behavior to be backward compatible.

My suggestion on this:
- We update the code to allow `otPlatResetToBootloader()` to return error and  `OT_ERROR_NOT_CAPABLE` to indicate this behavior is not supported (by RCP).  
- We would then  add a new `SPINEL_CAP_RESET_TO_BOOLADER` to let RCP indicate whether it provides this capability 
- And then implement the `otPlatResetToBoolLoader()` on posix host accordingly.

Reply:
@abtink Thanks for reviewing! I completely agree with your feedback and will make changes accordingly

Reply:
Agree with Abtin's suggestion. We should guarantee the backward compatible of the RCP/host behaviors.